src/sdk.rs

```diff
@@ -0,0 +1,283 @@
+use std::{env, time::Duration};
+use tonic::transport::Channel;
+
+use crate::proto::api::{self, sdk_client::SdkClient};
+
+#[cfg(feature = "player-tracking")]
+use crate::proto::alpha::{self, sdk_client::SdkClient as AlphaClient};
+
+pub use api::GameServer;
+
+pub type WatchStream = tonic::Streaming<GameServer>;
+
+use crate::errors::Result;
+
+#[inline]
+fn empty() -> api::Empty {
+    api::Empty {}
+}
+
+/// SDK is an instance of the Agones SDK
+#[derive(Clone)]
+pub struct Sdk {
+    client: SdkClient<Channel>,
+    #[cfg(feature = "player-tracking")]
+    alpha: AlphaClient<Channel>,
+}
+
+impl Sdk {
+    /// Starts a new SDK instance, and connects to localhost on the port specified
+    /// or else falls back to the `AGONES_SDK_GRPC_PORT` environment variable,
+    /// or defaults to 9357.
+    ///
+    /// # Errors
+    ///
+    /// - The port specified in `AGONES_SDK_GRPC_PORT` can't be parsed as a `u16`.
```

Comment:
It seems to not actually return an error here but fallback to 9357.

Reply:
Oh oops, good point.